WcaOnRails/db/migrate/20240206154319_migrate_seniors_to_roles.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class MigrateSeniorsToRoles < ActiveRecord::Migration[7.1]
+  def change
+    ActiveRecord::Base.transaction do
+      User.where(delegate_status: 'senior_delegate').each do |senior_delegate|
```

Comment:
Nit: you can use `find_each` instead of `each` here.
`each` is a general Ruby language construct that works on any array/list. But in case of Rails database, it requires loading all results completely into memory first, before it can iterate over them.
`find_each` is a Rails-specific method for database queries, which loops over the entries using SQL `LIMIT` and thus is much more efficient because it can load the user rows step by step while you're iterating over them.